File: backend/routers/focus.py
"""
focus.py - the Focus Reset: help with compulsive, high-stimulation habits.

Nothing here is shown unless the person's own entries carry a signal. The analysis
(routers/analyze.py) records reward-seeking behaviours an entry mentions: what, the
trigger, time of day, loss of control, the after-state and what was displaced. This
router turns those per-entry records into a 28-day picture and runs a guided programme
modelled on Anna Lembke's "DOPAMINE" structure (Data, Objectives, Problems, Abstinence,
Mindfulness, Insight, Next steps, Experiment) and Cameron Sepah's stimulus-control
"dopamine fasting": pick one behaviour, an abstinence window, self-binding rules and
replacement activities, then track urges and daily check-ins.

The app measures behaviour described in writing, never dopamine itself; the popular
name is used only because it is what people search for.
"""

from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session, joinedload
from datetime import datetime, timedelta, date as date_type
from typing import Literal, Optional
from pydantic import BaseModel, Field
from database import get_db
import models
from .auth import verify_session
import builtins

router = APIRouter(tags=["focus"])

WINDOW_DAYS = 28
RECENT_DAYS = 14
ABSTINENCE_OPTIONS = (7, 14, 30)


def _local_today(tz_offset: int) -> date_type:
    return (datetime.utcnow() + timedelta(minutes=tz_offset)).date()


def _parse_day(value: str) -> date_type:
    try:
        return datetime.strptime(value, "%Y-%m-%d").date()
    except ValueError:
        raise HTTPException(status_code=422, detail="date must be YYYY-MM-DD")


# --------------------------------------------------------------------------
# Signals from entries
# --------------------------------------------------------------------------

def _signal_days(user_id: str, db: Session, tz_offset: int) -> list[dict]:
    """Per local day in the window: the stimulation record of that day's entry, if any."""
    offset = timedelta(minutes=tz_offset)
    today = _local_today(tz_offset)
    window_start_utc = datetime.combine(today - timedelta(days=WINDOW_DAYS - 1), datetime.min.time()) - offset
    entries = (
        db.query(models.JournalEntry)
        .options(joinedload(models.JournalEntry.feedback))
        .filter(
            models.JournalEntry.user_id == user_id,
            models.JournalEntry.is_deleted == False,
            models.JournalEntry.date >= window_start_utc,
        )
        .all()
    )
    by_day: dict = {}
    for e in entries:
        fb = e.feedback
        if not fb or not e.date:
            continue
        d = (e.date + offset).date()
        s = fb.stimulation_data or {}
        by_day[d] = {
            "analysed": True,
            "load": int(s.get("load") or 0),
            "behaviours": s.get("behaviours") or [],
            "afterState": s.get("afterState") or "none",
            "lowMotivation": bool(s.get("lowMotivation")),
            "sleepDisrupted": bool(s.get("sleepDisrupted")),
            "craving": bool(s.get("cravingLanguage")),
            "displaced": s.get("displaced") or [],
        }
    days = []
    for i in builtins.range(WINDOW_DAYS - 1, -1, -1):
        d = today - timedelta(days=i)
        info = by_day.get(d)
        days.append({"date": d.isoformat(), **(info or {"analysed": False, "load": 0, "behaviours": [], "afterState": "none", "lowMotivation": False, "sleepDisrupted": False, "craving": False, "displaced": []})})
    return days


def _summarise(days: list[dict]) -> dict:
    signal = [d for d in days if d["load"] >= 1]
    recent = [d for d in days[-RECENT_DAYS:] if d["load"] >= 1]
    behaviours: dict = {}
    triggers: dict = {}
    time_of_day: dict = {}
    after: dict = {}
    displaced: list[str] = []
    lost_control_days = 0
    for d in signal:
        lost = False
        for b in d["behaviours"]:
            label = str(b.get("behaviour") or "").strip().lower()
            if not label:
                continue
            key = label
            entry = behaviours.setdefault(key, {"label": label, "category": b.get("category") or "other", "count": 0})
            entry["count"] += 1
            trig = str(b.get("trigger") or "").strip().lower()
            if trig and trig != "unclear":
                triggers[trig] = triggers.get(trig, 0) + 1
            tod = b.get("timeOfDay") or "unknown"
            time_of_day[tod] = time_of_day.get(tod, 0) + 1
            if b.get("lostControl"):
                lost = True
        if lost:
            lost_control_days += 1
        if d["afterState"] and d["afterState"] != "none":
            after[d["afterState"]] = after.get(d["afterState"], 0) + 1
        for item in d["displaced"]:
            if item and item not in displaced:
                displaced.append(item)
    top_behaviours = sorted(behaviours.values(), key=lambda b: -b["count"])[:6]
    top_triggers = [{"label": k, "count": v} for k, v in sorted(triggers.items(), key=lambda kv: -kv[1])[:5]]
    return {
        "analysedDays": sum(1 for d in days if d["analysed"]),
        "signalDays": len(signal),
        "recentSignalDays": len(recent),
        "heavyDays": sum(1 for d in signal if d["load"] >= 3),
        "lostControlDays": lost_control_days,
        "lowMotivationDays": sum(1 for d in days if d["lowMotivation"]),
        "sleepDisruptedDays": sum(1 for d in days if d["sleepDisrupted"]),
        "topBehaviours": top_behaviours,
        "topTriggers": top_triggers,
        "timeOfDay": time_of_day,
        "afterStates": after,
        "displaced": displaced[-8:],
    }


# --------------------------------------------------------------------------
# Plans, urges, check-ins
# --------------------------------------------------------------------------

class PlanCreate(BaseModel):
    behaviour: str = Field(min_length=2, max_length=120)
    category: Optional[str] = "other"
    objectives: Optional[str] = Field(default=None, max_length=2000)
    problems: Optional[str] = Field(default=None, max_length=2000)
    abstinenceDays: int = 7
    startDate: Optional[str] = None  # YYYY-MM-DD local; defaults to today in tz_offset
    rules: list[str] = Field(default_factory=list)
    replacements: list[str] = Field(default_factory=list)


class PlanUpdate(BaseModel):
    status: Optional[Literal["active", "completed", "abandoned"]] = None
    abstinenceDays: Optional[int] = None
    objectives: Optional[str] = Field(default=None, max_length=2000)
    problems: Optional[str] = Field(default=None, max_length=2000)
    rules: Optional[list[str]] = None
    replacements: Optional[list[str]] = None


class UrgeCreate(BaseModel):
    intensity: Optional[int] = Field(default=None, ge=1, le=5)
    acted: bool = False
    trigger: Optional[str] = Field(default=None, max_length=200)
    note: Optional[str] = Field(default=None, max_length=1000)


class CheckinUpsert(BaseModel):
    date: str
    urges: int = Field(default=0, ge=0, le=99)
    gaveIn: bool = False
    sleepOk: Optional[bool] = None
    note: Optional[str] = Field(default=None, max_length=1000)


def _active_plan(user_id: str, db: Session) -> Optional[models.FocusPlan]:
    return (
        db.query(models.FocusPlan)
        .filter(models.FocusPlan.user_id == user_id, models.FocusPlan.status == "active")
        .order_by(models.FocusPlan.created_at.desc())
        .first()
    )


def _serialize_plan(plan: models.FocusPlan, db: Session, tz_offset: int) -> dict:
    today = _local_today(tz_offset)
    start = _parse_day(plan.start_date)
    checkins = (
        db.query(models.FocusCheckin)
        .filter(models.FocusCheckin.plan_id == plan.id)
        .order_by(models.FocusCheckin.date.asc())
        .all()
    )
    urges = (
        db.query(models.FocusUrge)
        .filter(models.FocusUrge.plan_id == plan.id)
        .order_by(models.FocusUrge.logged_at.desc())
        .all()
    )
    by_date = {c.date: c for c in checkins}
    # Clean streak: consecutive checked-in days without giving in, counting back from today (or yesterday if today is unchecked).
    streak = 0
    cursor = today if today.isoformat() in by_date else today - timedelta(days=1)
    while cursor.isoformat() in by_date and not by_date[cursor.isoformat()].gave_in:
        streak += 1
        cursor -= timedelta(days=1)
    clean_days = sum(1 for c in checkins if not c.gave_in)
    day_number = (today - start).days + 1
    length = plan.abstinence_days or 7
    return {
        "id": plan.id,
        "behaviour": plan.behaviour,
        "category": plan.category,
        "objectives": plan.objectives,
        "problems": plan.problems,
        "abstinenceDays": length,
        "startDate": plan.start_date,
        "status": plan.status,
        "rules": plan.rules or [],
        "replacements": plan.replacements or [],
        "dayNumber": max(1, day_number),
        "daysLeft": max(0, length - day_number + 1),
        "finished": day_number > length,
        "cleanStreak": streak,
        "cleanDays": clean_days,
        "checkedInDays": len(checkins),
        "checkins": [
            {"date": c.date, "urges": c.urges or 0, "gaveIn": bool(c.gave_in), "sleepOk": c.sleep_ok, "note": c.note}
            for c in checkins
        ],
        "urges": {
            "total": len(urges),
            "surfed": sum(1 for u in urges if not u.acted),
            "recent": [
                {"id": u.id, "loggedAt": u.logged_at.isoformat() if u.logged_at else None, "intensity": u.intensity, "acted": bool(u.acted), "trigger": u.trigger, "note": u.note}
                for u in urges[:10]
            ],
        },
        "completedAt": plan.completed_at.isoformat() if plan.completed_at else None,
    }


@router.get("/api/focus/overview")
def get_overview(
    tz_offset: int = Query(default=0, ge=-840, le=840),
    user_id: str = Depends(verify_session),
    db: Session = Depends(get_db),
):
    """
    Everything the Focus pages need. `active` is false when the last two weeks of entries
    carry no stimulation signal and no plan is running, in which case the UI shows nothing.
    """
    days = _signal_days(user_id, db, tz_offset)
    summary = _summarise(days)
    plan = _active_plan(user_id, db)
    last_plan = None
    if not plan:
        previous = (
            db.query(models.FocusPlan)
            .filter(models.FocusPlan.user_id == user_id)
            .order_by(models.FocusPlan.created_at.desc())
            .first()
        )
        if previous:
            last_plan = {"behaviour": previous.behaviour, "status": previous.status, "completedAt": previous.completed_at.isoformat() if previous.completed_at else None}
    return {
        "success": True,
        "active": summary["recentSignalDays"] >= 1 or plan is not None,
        "windowDays": WINDOW_DAYS,
        "days": days,
        **summary,
        "plan": _serialize_plan(plan, db, tz_offset) if plan else None,
        "lastPlan": last_plan,
    }


@router.post("/api/focus/plans")
def create_plan(
    data: PlanCreate,
    tz_offset: int = Query(default=0, ge=-840, le=840),
    user_id: str = Depends(verify_session),
    db: Session = Depends(get_db),
):
    if data.abstinenceDays not in ABSTINENCE_OPTIONS:
        raise HTTPException(status_code=422, detail=f"abstinenceDays must be one of {ABSTINENCE_OPTIONS}")
    start = _parse_day(data.startDate) if data.startDate else _local_today(tz_offset)
    # One plan at a time: starting a new one closes the previous.
    for old in db.query(models.FocusPlan).filter(models.FocusPlan.user_id == user_id, models.FocusPlan.status == "active").all():
        old.status = "abandoned"
        old.completed_at = datetime.utcnow()
    plan = models.FocusPlan(
        user_id=user_id,
        behaviour=data.behaviour.strip(),
        category=data.category or "other",
        objectives=(data.objectives or "").strip() or None,
        problems=(data.problems or "").strip() or None,
        abstinence_days=data.abstinenceDays,
        start_date=start.isoformat(),
        status="active",
        rules=[r.strip() for r in data.rules if r.strip()][:12],
        replacements=[r.strip() for r in data.replacements if r.strip()][:12],
    )
    db.add(plan)
    db.commit()
    db.refresh(plan)
    return {"success": True, "plan": _serialize_plan(plan, db, tz_offset)}


@router.patch("/api/focus/plans/{plan_id}")
def update_plan(
    plan_id: str,
    data: PlanUpdate,
    tz_offset: int = Query(default=0, ge=-840, le=840),
    user_id: str = Depends(verify_session),
    db: Session = Depends(get_db),
):
    plan = db.query(models.FocusPlan).filter(models.FocusPlan.id == plan_id, models.FocusPlan.user_id == user_id).first()
    if not plan:
        raise HTTPException(status_code=404, detail="Plan not found")
    if data.abstinenceDays is not None:
        if data.abstinenceDays not in ABSTINENCE_OPTIONS or data.abstinenceDays < (plan.abstinence_days or 0):
            raise HTTPException(status_code=422, detail="abstinenceDays can only extend the plan to 7, 14 or 30 days")
        plan.abstinence_days = data.abstinenceDays
    if data.objectives is not None:
        plan.objectives = data.objectives.strip() or None
    if data.problems is not None:
        plan.problems = data.problems.strip() or None
    if data.rules is not None:
        plan.rules = [r.strip() for r in data.rules if r.strip()][:12]
    if data.replacements is not None:
        plan.replacements = [r.strip() for r in data.replacements if r.strip()][:12]
    if data.status is not None and data.status != plan.status:
        plan.status = data.status
        plan.completed_at = datetime.utcnow() if data.status in ("completed", "abandoned") else None
    db.commit()
    db.refresh(plan)
    return {"success": True, "plan": _serialize_plan(plan, db, tz_offset)}


@router.post("/api/focus/urges")
def log_urge(data: UrgeCreate, user_id: str = Depends(verify_session), db: Session = Depends(get_db)):
    plan = _active_plan(user_id, db)
    urge = models.FocusUrge(
        user_id=user_id,
        plan_id=plan.id if plan else None,
        intensity=data.intensity,
        acted=data.acted,
        trigger=(data.trigger or "").strip() or None,
        note=(data.note or "").strip() or None,
    )
    db.add(urge)
    db.commit()
    db.refresh(urge)
    return {"success": True, "urge": {"id": urge.id, "acted": urge.acted, "intensity": urge.intensity, "loggedAt": urge.logged_at.isoformat() if urge.logged_at else None, "planId": urge.plan_id}}


@router.post("/api/focus/checkins")
def upsert_checkin(
    data: CheckinUpsert,
    tz_offset: int = Query(default=0, ge=-840, le=840),
    user_id: str = Depends(verify_session),
    db: Session = Depends(get_db),
):
    plan = _active_plan(user_id, db)
    if not plan:
        raise HTTPException(status_code=409, detail="No active plan")
    day = _parse_day(data.date)
    if day < _parse_day(plan.start_date) or day > _local_today(tz_offset):
        raise HTTPException(status_code=422, detail="date must be within the plan and not in the future")
    checkin = db.query(models.FocusCheckin).filter(models.FocusCheckin.plan_id == plan.id, models.FocusCheckin.date == data.date).first()
    if not checkin:
        checkin = models.FocusCheckin(user_id=user_id, plan_id=plan.id, date=data.date)
        db.add(checkin)
    checkin.urges = data.urges
    checkin.gave_in = data.gaveIn
    checkin.sleep_ok = data.sleepOk
    checkin.note = (data.note or "").strip() or None
    db.commit()
    db.refresh(plan)
    return {"success": True, "plan": _serialize_plan(plan, db, tz_offset)}

File: frontend/src/lib/focus.ts
/**
 * Focus Reset: shared types and programme content.
 *
 * The programme follows Anna Lembke's DOPAMINE structure (Data, Objectives, Problems,
 * Abstinence, Mindfulness, Insight, Next steps, Experiment) and Cameron Sepah's
 * stimulus-control "dopamine fasting". The app measures behaviour described in the
 * person's own writing, never dopamine itself.
 */

export interface SignalBehaviour {
  behaviour: string
  category: string
  trigger: string
  timeOfDay: string
  lostControl: boolean
}

export interface SignalDay {
  date: string
  analysed: boolean
  load: number
  behaviours: SignalBehaviour[]
  afterState: string
  lowMotivation: boolean
  sleepDisrupted: boolean
  craving: boolean
  displaced: string[]
  // mind fitness
  fog: boolean
  fogNote: string
  minutes: number
  shortForm: boolean
  builders: string[]
  rot: number
}

export interface MindDay {
  date: string
  analysed: boolean
  rot: number
  fog: boolean
  shortForm: boolean
  minutes: number
  builders: string[]
  manual: string[]
}

export interface MindGuide {
  startDate: string
  day: number
  week: number
  finished: boolean
}

export interface MindSummary {
  active: boolean
  fogDays: number
  recentFogDays: number
  shortFormDays: number
  rotDays: number
  heavyRotDays: number
  avgMinutes: number | null
  totalMinutes: number
  notes: string[]
  builders: Record<string, { weekDays: number; monthDays: number; target: number }>
  weekScore: number
  weekBuilderDays: number
  days: MindDay[]
  guide: MindGuide | null
}

export interface FocusPlan {
  id: string
  behaviour: string
  category: string | null
  objectives: string | null
  problems: string | null
  abstinenceDays: number
  startDate: string
  status: "active" | "completed" | "abandoned"
  rules: string[]
  replacements: string[]
  dayNumber: number
  daysLeft: number
  finished: boolean
  cleanStreak: number
  cleanDays: number
  checkedInDays: number
  checkins: { date: string; urges: number; gaveIn: boolean; sleepOk: boolean | null; note: string | null }[]
  urges: { total: number; surfed: number; recent: { id: string; loggedAt: string | null; intensity: number | null; acted: boolean; trigger: string | null; note: string | null }[] }
  completedAt: string | null
}

export interface FocusOverview {
  success: boolean
  active: boolean
  windowDays: number
  days: SignalDay[]
  analysedDays: number
  signalDays: number
  recentSignalDays: number
  heavyDays: number
  lostControlDays: number
  lowMotivationDays: number
  sleepDisruptedDays: number
  topBehaviours: { label: string; category: string; count: number }[]
  topTriggers: { label: string; count: number }[]
  timeOfDay: Record<string, number>
  afterStates: Record<string, number>
  displaced: string[]
  plan: FocusPlan | null
  lastPlan: { behaviour: string; status: string; completedAt: string | null } | null
  mind: MindSummary
}

export const CATEGORY_LABELS: Record<string, string> = {
  screens: "Screens",
  social_media: "Social media",
  video: "Video",
  gaming: "Gaming",
  porn: "Porn",
  gambling: "Gambling",
  food: "Food",
  shopping: "Shopping",
  substances: "Substances",
  other: "Other",
}

/** Categories where an app is not enough and a person should be involved. */
export const NEEDS_SUPPORT = new Set(["substances", "gambling"])

export const OBJECTIVE_OPTIONS = ["boredom", "loneliness", "avoiding a task", "winding down", "falling asleep", "stress relief", "habit, no reason"]

/** Self-binding rule suggestions, in Lembke's three flavours: physical, chronological, categorical. */
export const RULE_SUGGESTIONS: Record<string, string[]> = {
  screens: ["Phone charges outside the bedroom", "No screens after 10pm", "Home screen holds only tools, no feeds", "Greyscale display after dinner"],
  social_media: ["Log out after each use, no saved password", "Apps deleted from the phone; browser only", "Two fixed check-in times a day, 10 minutes each", "Phone charges outside the bedroom"],
  video: ["Autoplay off everywhere", "Watch only at the table, never in bed", "One episode, decided before pressing play", "No screens after 10pm"],
  gaming: ["Console or client uninstalled for the window", "Play only on named days, with a timer set first", "Controller lives in a drawer in another room"],
  porn: ["Content filter on every device", "Phone charges outside the bedroom", "No devices in bed"],
  gambling: ["Self-exclusion registered with each site", "Cards and cash held by someone you trust", "Blocking software installed"],
  food: ["Trigger foods not kept in the house", "Eat only at the table, never with a screen", "A planned snack replaces the impulse one"],
  shopping: ["Cards removed from browsers and apps", "A 48-hour list before any purchase", "Unsubscribe from every sale email"],
  substances: ["None kept at home", "Tell one person the window and the rule", "Avoid the places where it happens"],
  other: ["Remove the cue from sight", "Set a fixed time and a timer", "Tell one person the rule"],
}

export const REPLACEMENT_SUGGESTIONS = ["A ten-minute walk", "A page of writing here", "Stretching", "Cold water on the face", "A call to a friend", "Reading, paper only", "A shower", "Tidying one surface"]

export const PROGRAMME_STEPS = [
  { key: "data", letter: "D", title: "Data", blurb: "What, how much, when. Your entries already tell part of this story." },
  { key: "objectives", letter: "O", title: "Objectives", blurb: "What the behaviour does for you. It always does something." },
  { key: "problems", letter: "P", title: "Problems", blurb: "What it costs, in your own words." },
  { key: "abstinence", letter: "A", title: "Abstinence", blurb: "One behaviour, one window. Long enough for the balance to reset." },
  { key: "mindfulness", letter: "M", title: "Mindfulness", blurb: "Urges peak and pass. You learn to watch one go by." },
  { key: "insight", letter: "I", title: "Insight", blurb: "Keep writing daily. The pattern shows itself." },
  { key: "next", letter: "N", title: "Next steps", blurb: "Rules that bind you in advance, when you are calm." },
  { key: "experiment", letter: "E", title: "Experiment", blurb: "Run it, track it, adjust it." },
]

export function tzOffset() {
  return -new Date().getTimezoneOffset()
}

export function localToday() {
  const d = new Date()
  return `${d.getFullYear()}-${String(d.getMonth() + 1).padStart(2, "0")}-${String(d.getDate()).padStart(2, "0")}`
}

export async function fetchOverview(): Promise<FocusOverview | null> {
  try {
    const res = await fetch(`/api/focus/overview?tz_offset=${tzOffset()}`)
    if (!res.ok) return null
    const data = await res.json()
    return data?.success ? data : null
  } catch {
    return null
  }
}

// ---------------------------------------------------------------- mind fitness

export const GUIDE_DAYS = 28

/** Brain-building activities, with how many days a week each is worth aiming for. Order is display order. */
export const BUILDERS: { key: string; label: string; target: number; example: string; why: string }[] = [
  { key: "sleep", label: "Slept enough", target: 5, example: "Seven hours or more", why: "Sleep is when the day's memories are filed. Short nights show up first as poor attention." },
  { key: "deep_reading", label: "Read long-form", target: 4, example: "Twenty minutes of a book or a long article, no skimming", why: "Sustained reading trains the same attention that feeds wear down, a fragment at a time." },
  { key: "deep_work", label: "Deep work block", target: 3, example: "One task, no tabs, twenty-five to fifty minutes", why: "Focus is a capacity. Single-task blocks rebuild it the way sets rebuild a muscle." },
  { key: "exercise", label: "Moved hard", target: 3, example: "A run, a lift, a fast walk", why: "Exercise raises BDNF, a growth factor for neurons, and reliably improves memory and executive function." },
  { key: "rest", label: "Real rest", target: 3, example: "Boredom, a nap, staring out of a window, no feed", why: "Unfilled minutes let the mind wander, which is where consolidation and new ideas happen." },
  { key: "nature", label: "Time outdoors", target: 2, example: "A walk somewhere green", why: "Attention Restoration Theory: natural settings let directed attention recover." },
  { key: "learning", label: "Learned something hard", target: 2, example: "A language, an instrument, a proof, a recipe from scratch", why: "Effortful learning builds new connections and cognitive reserve; easy content builds nothing." },
  { key: "conversation", label: "Real conversation", target: 2, example: "Face to face, or a proper call", why: "Live conversation works memory, attention and empathy at the same time." },
  { key: "creating", label: "Made something", target: 1, example: "Wrote, drew, built, cooked, played music", why: "Making reverses the direction of the habit: output instead of intake." },
  { key: "play", label: "Played", target: 1, example: "A board game, a sport, a puzzle with someone", why: "Play is problem-solving at low stakes; it keeps thinking flexible." },
]

export const BUILDER_LABELS: Record<string, string> = Object.fromEntries(BUILDERS.map(b => [b.key, b.label]))

/** Four weeks, one theme each: subtract, rebuild, feed, keep. Everything here is something to do, nothing to buy. */
export const MIND_GUIDE_WEEKS: { week: number; title: string; theme: string; why: string; practices: string[]; prompt: string }[] = [
  {
    week: 1, title: "Notice and subtract", theme: "See how much of the day is passive, and take the worst of it away.",
    why: "The research on short-form video and attention is young but consistent: heavier use goes with poorer attention control and working memory, and the association weakens when use drops. Week one is measurement and one subtraction.",
    practices: [
      "Write the minutes of passive scrolling into each entry, even roughly. The app keeps the number.",
      "Turn off autoplay everywhere and move short-video apps off the home screen, or delete them for the month.",
      "The phone charges outside the bedroom. Nothing on it in the first thirty minutes after waking.",
      "Once a day, wait somewhere without the phone: a queue, a lift, a kettle. Let it be boring.",
    ],
    prompt: "When did the fog come today, and what had I just been doing?",
  },
  {
    week: 2, title: "Rebuild attention", theme: "Train the capacity directly, and restore it between sessions.",
    why: "Attention responds to training. Long-form reading and single-task blocks rebuild it; sleep and green space restore it. None of this needs equipment.",
    practices: [
      "Twenty minutes of a paper book or a long article every day, no skimming, phone in another room.",
      "One deep-work block of twenty-five minutes: one task, notifications off. Add five minutes each day it goes well.",
      "A walk somewhere green, twice this week, without earphones.",
      "Seven hours in bed on five nights of the seven. Sleep is where the day gets filed.",
    ],
    prompt: "Where did my attention go today without being pulled?",
  },
  {
    week: 3, title: "Feed the brain", theme: "Put in the inputs with the best evidence, and make instead of consume.",
    why: "Exercise, effortful learning and live conversation carry the strongest evidence for memory and executive function. Making something reverses the intake habit.",
    practices: [
      "Move hard three times: a run, a lift, anything that leaves you breathing.",
      "Pick one hard thing to learn and give it twenty minutes on two days: an instrument, a language, a skill by hand.",
      "One real conversation, face to face or a proper call, with no screen in it.",
      "Make one thing this week instead of consuming: write, draw, cook from scratch, build.",
    ],
    prompt: "What did I make or learn today that did not exist this morning?",
  },
  {
    week: 4, title: "Make it stick", theme: "Let the environment do the work, so the habits hold without willpower.",
    why: "Habits last when the cue is gone and the replacement is easy. This week compares week one's entries with now and turns what helped into rules.",
    practices: [
      "Read the entries from week one. Count the fog days then and now.",
      "Choose a daily passive-consumption budget you can keep, and one time of day when feeds are fine.",
      "Keep the three builders that made the biggest difference; drop the rest without guilt.",
      "Write the rules into an entry, as if to a friend who is starting week one.",
    ],
    prompt: "What has changed in how my mind feels, and what do I keep?",
  },
]

export const BRAIN_ROT_NOTES = [
  "\"Brain rot\" was Oxford's word of the year for 2024: the supposed deterioration of a person's mental or intellectual state from overconsumption of trivial online content. It is a description, not a diagnosis.",
  "What the studies show so far: heavier short-form video use goes with poorer attention control, working memory and learning, and the association weakens when use drops. The work is recent and mostly correlational, so read it as a strong hint rather than a verdict.",
  "The reverse is better established. Sleep, exercise, long-form reading, effortful learning, time in nature and live conversation each have decades of evidence behind them for attention and memory. That is why the builders are what they are.",
  "This app reads what you write about your attention. It does not test your brain. Fog that is persistent, heavy or new is a conversation for a doctor, not an app.",
]
